apps/query-demo/querydemo/Query_Demo.py

```diff
@@ -364,43 +365,97 @@ def show(self):
 
         with st.container(border=True):
             st.write(f"`{self.path}`")
-            tab1, tab2 = st.tabs(["PDF", "Metadata"])
-            with tab1:
-                col1, col2, col3, col4 = st.columns(4)
-                with col1:
-                    if self.chat_message.button("First", use_container_width=True):
-                        st.session_state.pagenum = 1
-                with col2:
-                    if self.chat_message.button("Prev", use_container_width=True):
-                        st.session_state.pagenum = max(1, st.session_state.pagenum - 1)
-                with col3:
-                    if self.chat_message.button("Next", use_container_width=True):
-                        st.session_state.pagenum += 1
-                with col4:
-                    self.chat_message.download_button(
-                        "Download", content, os.path.basename(self.path), "pdf", use_container_width=True
-                    )
-                pdf_viewer(content, pages_to_render=[st.session_state.pagenum], key=self.chat_message.next_key())
-
+            encoded = base64.b64encode(content).decode("utf-8")
+            pdf_display = f'<iframe src="data:application/pdf;base64,{encoded}" width="600" height="800" type="application/pdf"></iframe>'
+            st.markdown(pdf_display, unsafe_allow_html=True)
 
-# TODO: Fetch underlying Sycamore doc and show metadata here.
-#            with tab2:
-#                props = {k: v for k, v in doc.properties["entity"].items() if v is not None}
-#                st.dataframe(props)
 
-
-def query_data_source(query: str, index: str) -> Tuple[Any, LogicalPlan]:
-    """Run a Sycamore query."""
-    sqclient = SycamoreQueryClient(
-        s3_cache_path=st.session_state.s3_cache_path if st.session_state.use_cache else None,
-    )
-    with st.spinner("Generating plan..."):
-        plan = generate_plan(sqclient, query, index)
-        with st.expander("Query plan"):
-            st.write(plan)
-    with st.spinner("Running Sycamore query..."):
-        st.session_state.query_id, result = run_plan(sqclient, plan)
-    return result, plan
+def get_embedding_model_id() -> str:
+    """Get the embedding model ID from the OpenSearch instance."""
+    query = {
+        "query": {
+            "bool": {
+                "must": [
+                    {
+                        "match": {"name": "all-MiniLM-L6-v2"},
```

Comment:
this is fine for now but we should use a better model when we actually demo for the comparison to at least be sorta fair.

Reply:
This is just the embedding model, though, right? IIUC it is using OpenAI for the actual RAG step, no?